src/ffi/dial_ffi.rs

```diff
@@ -137,6 +141,21 @@ pub unsafe extern "C" fn dial(
         ur
     };
     let allow_insec = c_allow_insec;
+    if!c_payload.is_null() && c_type.is_null(){
+        println!("Error missing credential: secret type");
```

Comment:
I don't think we should be using `println` in production code. Can we do error logging instead?

Probably would be good to replace any existing `println` lines in this file with `log` statements as well, while we're at it.

Reply:
Sorry I missed this, thank you for the reminder! It will be updated in the next commit.